### PR TITLE
change(debug): move 'enable_debug' form config.yaml to debug.yaml

### PR DESCRIPTION
**Pull request by tzssangglass**
### What this PR does / why we need it:

move 'enable_debug' form config.yaml to debug.yaml, make the configuration of debug mode clear and central

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by tokers — APPROVED** [review]

File: docs/zh/latest/architecture-design/debug-mode.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ title: Debug Mode
 
 ### 基本调试模式
 
-设置 `conf/config.yaml` 中的 `apisix.enable_debug` 为 `true`，即可开启基本调试模式。
+设置 `conf/debug.yaml` 中的 `enable_debug` 为 `true`，即可开启基本调试模式。
```
Comment (leslie-tsang) [inline docs/zh/latest/architecture-design/debug-mode.md:26]: Hi, for those who using apisix versions before 2.9.0, this document may be confused

Comment (spacewander) [reply docs/zh/latest/architecture-design/debug-mode.md:26]: We can add a note for the version before 2.10.0?

File: conf/debug.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+enable_debug: false
```
Comment (spacewander) [inline conf/debug.yaml:17]:
```suggestion
basic:
  enable: false
```
would be better?

Comment (tzssangglass) [reply conf/debug.yaml:17]: Looks better, but this `enable` will conflict with https://github.com/apache/apisix/blob/b52f95b53c689cfd2fd0f4cbf0fefedd4cce9b5e/t/debug/hook.t#L34

I don't know how to modify this with perl.

Comment (spacewander) [reply conf/debug.yaml:17]: Let me do it for you 😃

**Review by spacewander — APPROVED** [review]

**Pull request merged**